### PR TITLE
refactor(sdk): Expand glob before passing to internal process

### PR DESCRIPTION
**Pull request by timoffex**
Resolves WB-18094 by expanding globs in `run.save()`. There may still exist `FilesRecords` with globs, which the Nexus will not process correctly.

### CODE REVIEW

File: wandb/sdk/wandb_run.py
Comment (kptkin) [inline wandb/sdk/wandb_run.py:1]: i'm surprised that we don't cover all these

**Comment by kptkin** [comment]
could you explain more how it diverges from the python version?

File: wandb/sdk/wandb_run.py
Comment (timoffex) [reply wandb/sdk/wandb_run.py:1]: that looks wrong... should definitely be covered. The codecov comment says it is: https://github.com/wandb/wandb/pull/7235#issuecomment-2021723139

**Comment by timoffex** [comment]
> could you explain more how it diverges from the python version?

`run.save()` should work the same in both. However, there may be other places where FilesRecords with globs are created and passed to the internal process. What I've found:

* `Run._datatypes_callback`, which can probably just call `save` instead
* The `tb_watcher.py`, which I think is only used with the Python service implementation
* In `system_info.py`; this doesn't seem to pass a glob though, so it might be fine

That's just the Python code that calls `publish_files`, but maybe there is also Go code that creates FilesRecords with globs, or other Python codepaths.

**Comment by kptkin** [comment]
> > could you explain more how it diverges from the python version?
> 
> `run.save()` should work the same in both. However, there may be other places where FilesRecords with globs are created and passed to the internal process. What I've found:
> 
> * `Run._datatypes_callback`, which can probably just call `save` instead
> * The `tb_watcher.py`, which I think is only used with the Python service implementation
> * In `system_info.py`; this doesn't seem to pass a glob though, so it might be fine
> 
> That's just the Python code that calls `publish_files`, but maybe there is also Go code that creates FilesRecords with globs, or other Python codepaths.


- system_info.py -is also used with the python service it is part of the system monitor service.
- Run._datatypes_callback - yep, but i think it always a concrete path

**Review by kptkin — APPROVED** [review]

**Pull request merged**